### PR TITLE
Add artifactId to service events

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Add the artifactId to some of the service events:
- deployed
- upgraded
- rolledback

The model is that one service is associated to one artifact. As already documented in the primer.md, this is a simplified model and it may change in future versions.

The format of the artifactId in the examples is consistent with the rest of the spec for now. All artifactIds will align to the purl format in an upcoming pull request.

Fixes: #58

Signed-off-by: Andrea Frittoli <andrea.frittoli@gmail.com>

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has the [primer doc](https://github.com/cdevents/spec/blob/main/primer.md) been updated if a design decision is involved
- [x] Have the [JSON schemas](https://github.com/tektoncd/community/blob/main/standards.md#tests) been updated if the specification changed
- [x] Has spec version and event versions been updated according to the [versioning policy](https://github.com/cdevents/spec/blob/main/primer.md#versioning)
- [x] Meets the [CDEvents contributor standards](https://github.com/cdevents/.github/blob/main/docs/CONTRIBUTING.md)

### CODE REVIEW

File: continuous-deployment-pipeline-events.md [hunk 1]
```diff
@@ -41,6 +41,7 @@ A `service` can represent for example a binary that is running, a daemon, an app
 | id    | `String` | Uniquely identifies the subject within the source. | `service/myapp`, `daemonset/myapp` |
 | source | `URI-Reference` | [source](../spec.md#source) from the context | `staging/tekton`, `tekton-dev-123`|
 | environmentId | `String` | Id of the environment where the service runs | `dev`, `staging`, `production`, `ci-123`|
+| artifactId | `String` | Identifier of the artifact deployed with this service |  `0b31b1c02ff458ad9b7b81cbdf8f028bd54699fa151f221d1e8de6817db93427`, `927aa808433d17e315a258b98e2f1a55f8258e0cb782ccb76280646d0dbe17b5`, `six-1.14.0-py2.py3-none-any.whl` |
```
Comment (afrittoli) [inline continuous-deployment-pipeline-events.md:44]: I will update the format of artifact IDs in an upcoming PR to address https://github.com/cdevents/spec/issues/38 (once this one is merged)

**Comment by e-backmark-ericsson** [comment]
The PR comment should say "deployed" instead of "published", right?

**Comment by afrittoli** [comment]
> The PR comment should say "deployed" instead of "published", right?

Nice catch, thank you, I will update the commit message

**Comment by e-backmark-ericsson** [comment]
Why isn't the artifactId a mandatory contents parameter? What use is it to send a service deployed event if it doesn't reference what is deployed?

**Comment by afrittoli** [comment]
> Why isn't the artifactId a mandatory contents parameter? What use is it to send a service deployed event if it doesn't reference what is deployed?

I was being conservative in adding new mandatory fields, but I'd be happy to switch that to mandatory

**Comment by e-backmark-ericsson** [comment]
I'd say it should be mandatory in 0.1. We could still change that in 0.2 or later (but not after 1.0 unless stepping the major version). I think we might have other major changes coming down the line before we reach 1.0 so it shouldn't be a huge drawback if we change our minds on this one later.

**Comment by afrittoli** [comment]
> I'd say it should be mandatory in 0.1. We could still change that in 0.2 or later (but not after 1.0 unless stepping the major version). I think we might have other major changes coming down the line before we reach 1.0 so it shouldn't be a huge drawback if we change our minds on this one later.

Updated, it's mandatory now

**Review by e-backmark-ericsson — APPROVED** [review]

**Pull request merged**